pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1281,13 +1281,46 @@ func (self *AWSCloud) TCPLoadBalancerExists(name, region string) (bool, error) {
 	return false, nil
 }
 
+// Retrieves instance's vpc id from metadata
+func (self *AWSCloud) findVPCId() (string, error) {
+
+	metadata := self.awsServices.Metadata()
+	macs_bytes, err := metadata.GetMetaData("network/interfaces/macs/")
+	if err != nil {
+		err = fmt.Errorf("Could not list interfaces of the instance", err)
+		return "", err
+	}
+
+	// loop over interfaces, first vpc id returned wins
+	for _, mac_path := range strings.Split(string(macs_bytes), "\n") {
+		if len(mac_path) == 0 {
+			continue
+		}
+		url := fmt.Sprintf("network/interfaces/macs/%svpc-id", mac_path)
+		vpc_id_bytes, _ := metadata.GetMetaData(url)
```

Comment:
I think you want to s/_/err here, because otherwise err on the next line is checking something unrelated